pkg/inventory/inventory-client.go

```diff
@@ -411,19 +397,16 @@ func (cic *ClusterInventoryClient) applyInventoryObj(obj *unstructured.Unstructu
 	if obj == nil {
 		return fmt.Errorf("attempting apply a nil inventory object")
 	}
-	invInfo, err := cic.toInfo(obj)
-	if err != nil {
-		return err
-	}
-	helper := cic.helperFromInfo(invInfo)
-	klog.V(4).Infof("replacing inventory object: %s/%s", invInfo.Namespace, invInfo.Name)
-	var overwrite = true
-	replacedObj, err := helper.Replace(invInfo.Namespace, invInfo.Name, overwrite, invInfo.Object)
+
+	mapping, err := cic.mapper.RESTMapping(obj.GroupVersionKind().GroupKind())
```

Comment:
Here and in other places where `RESTMapping()` is being called I think we should get the mapping for the version of the object, not the default version.
```suggestion
	gvk := obj.GroupVersionKind()
	mapping, err := cic.mapper.RESTMapping(gvk.GroupKind(), gvk.Version())
```

Reply:
Agree. I have updated the PR.